### ISSUE TITLE
Onboarding Wizard error in Jetpack step

### ISSUE THREAD

**Issue by deltaWhiskey** [woocommerce/woocommerce#25651]
**Describe the bug**
When going through the onboarding wizard, when I click the "Continue with Jetpack" button, I get a blank white page with this message:

```
Fatal error: Uncaught Error: Call to a member function register() on null in /srv/users/userfca1304d/apps/userfca1304d/public/wp-content/plugins/jetpack/class.jetpack.php:5260

Stack trace:
#0 /srv/users/userfca1304d/apps/userfca1304d/public/wp-content/plugins/jetpack/class.jetpack.php(3361): Jetpack::register()

#1 /srv/users/userfca1304d/apps/userfca1304d/public/wp-content/plugins/woocommerce/includes/admin/class-wc-admin-setup-wizard.php(2404): Jetpack::try_registration()

#2 /srv/users/userfca1304d/apps/userfca1304d/public/wp-content/plugins/woocommerce/includes/admin/class-wc-admin-setup-wizard.php(326): WC_Admin_Setup_Wizard->wc_setup_activate_save(Object(WC_Admin_Setup_Wizard))

#3 /srv/users/userfca1304d/apps/userfca1304d/public/wp-includes/class-wp-hook.php(288): WC_Admin_Setup_Wizard->setup_wizard('')

#4 /srv/users/userfca1304d/apps/userfca1304d/public/wp-includes/class-wp-hook.php(312): WP_Hook->apply_filters(NULL, Array)

#5 /srv/users/userfca1304d/apps/userfca1304d/public/wp-includes/plugin.php(478): WP_Hook->do_a in /srv/users/userfca1304d/apps/userfca1304d/public/wp-content/plugins/jetpack/class.jetpack.php on line 5260
```

Clicking refresh takes me to the expected Jetpack connection flow.

**To Reproduce**
1. Fresh WordPress 5.3.2 install with WooCommerce 3.9.1. No other plugins installed. Theme is "Twenty Twenty".
2. Go through onboarding wizard (see screenshots for the options I chose.)
3. Click the "Continue with Jetpack" button, and you'll see the error message.

When I go through this process, stop at the error message page, then check the plugins page, I see that Jetpack 8.2 is installed and activated.

**Screenshots**
See first comment.

**Expected behavior**
Clicking "Continue with Jetpack" takes me to the Jetpack connection flow.

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**WordPress Environment**
<details>
```
`
### WordPress Environment ###

WordPress address (URL): https://original-pilot.jurassic.ninja
Site address (URL): https://original-pilot.jurassic.ninja
WC Version: 3.9.1
REST API Version: ✔ 1.0.7
Log Directory Writable: ✔
WP Version: 5.3.2
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.41 (Unix) OpenSSL/1.0.2g
PHP Version: 7.3.14
PHP Post Max Size: 1 GB
PHP Time Limit: 30
PHP Max Input Vars: 5000
cURL Version: 7.47.0
OpenSSL/1.0.2g

SUHOSIN Installed: –
MySQL Version: 5.7.29-0ubuntu0.16.04.1-log
Max Upload Size: 512 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 3.9.1
WC Database Prefix: wp_56fda88b3b_
Total Database Size: 3.06MB
Database Data Size: 2.08MB
Database Index Size: 0.98MB
wp_56fda88b3b_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_56fda88b3b_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_56fda88b3b_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_56fda88b3b_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_56fda88b3b_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_56fda88b3b_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_56fda88b3b_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_56fda88b3b_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_56fda88b3b_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_56fda88b3b_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_56fda88b3b_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_56fda88b3b_options: Data: 1.50MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_postmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_56fda88b3b_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_56fda88b3b_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_56fda88b3b_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_56fda88b3b_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_56fda88b3b_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_56fda88b3b_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB

### Post Type Counts ###

attachment: 1
page: 6
post: 2

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (3) ###

Companion Plugin: by Osk – 1.11
Jetpack by WordPress.com: by Automattic – 8.2
WooCommerce: by Automattic – 3.9.1

### Inactive Plugins (0) ###


### Settings ###

API Enabled: –
Force SSL: –
Currency: GBP (£)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –

### WC Pages ###

Shop base: #6 - /shop/
Cart: #7 - /cart/
Checkout: #8 - /checkout/
My account: #9 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty
Version: 1.1
Author URL: https://wordpress.org/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 0
Oldest: –
Newest: –

Pending: 0
Oldest: –
Newest: –

Canceled: 0
Oldest: –
Newest: –

In-progress: 0
Oldest: –
Newest: –

Failed: 0
Oldest: –
Newest: –

`
```
</details>

**Comment by deltaWhiskey** [comment]
Screen shots, showing the options I picked:

<img width="729" alt="Screen Shot 2020-02-12 at 12 08 36 PM" src="https://user-images.githubusercontent.com/15386509/74374326-09e8bb80-4d93-11ea-925d-3036cde68cf0.png">

<img width="739" alt="Screen Shot 2020-02-12 at 12 08 49 PM" src="https://user-images.githubusercontent.com/15386509/74374338-0ce3ac00-4d93-11ea-80da-533d165b6e22.png">

<img width="724" alt="Screen Shot 2020-02-12 at 12 08 58 PM" src="https://user-images.githubusercontent.com/15386509/74374345-0f460600-4d93-11ea-8d3f-52846744e5c3.png">

<img width="727" alt="Screen Shot 2020-02-12 at 12 09 08 PM" src="https://user-images.githubusercontent.com/15386509/74374351-11a86000-4d93-11ea-815a-4b5065c1e18d.png">

<img width="727" alt="Screen Shot 2020-02-12 at 12 09 25 PM" src="https://user-images.githubusercontent.com/15386509/74374354-13722380-4d93-11ea-94ed-bea46990a91e.png">

**Comment by tammullen** [comment]
Hi @deltaWhiskey 

Thank you for reporting this, much appreciated. 
I can reproduce it on my end using JN without any existing Jetpack install in both 3.92 and 4.0 beta

**Comment by rodrigoprimo** [comment]
I was also able to reproduce this issue. It happens when Jetpack is not installed or when it is installed but inactive.

It seems that the OBW is calling `Jetpack::try_registration()` (https://github.com/woocommerce/woocommerce/blob/master/includes/admin/class-wc-admin-setup-wizard.php#L2431) before `Jetpack::configure()` runs. The latter is the method that sets `$this->connection_manager` (https://github.com/Automattic/jetpack/blame/master/class.jetpack.php#L713) which is the property returned by `Jetpack::connection()`. That is why is `$connection` is `null` (https://github.com/Automattic/jetpack/blame/master/class.jetpack.php#L5246) and this causes the error mentioned in the description of this issue. I'm not sure when this started happening and if it started because of a change in WooCommerce or Jetpack.

**Comment by rodrigoprimo** [comment]
I did a little bit more testing and found that it was a change that was introduced in Jetpack 8.2 that is causing this error. I tried using `git bisect` in the Jetpack repository to determine exactly what commit caused the problem but didn't succeed since they seem to use some build mechanism and the commit that I found seems to include all the changes (or several changes) that were included in the 8.2 version: https://github.com/Automattic/jetpack/commit/a45c0d3e14659cb6f79c70730f560f23eaa223d0

It is possible to work around this fatal error by adding the following code before calling `Jetpack::try_registration()` (https://github.com/woocommerce/woocommerce/blob/master/includes/admin/class-wc-admin-setup-wizard.php#L2431):

```php
$jetpack = Jetpack::init();
$jetpack->configure();
```

I'm not sure what are the consequences of manually initializing and configuring Jetpack here and if we should do it. I will try to check with someone from the Jetpack team before proceeding.

**Comment by rodrigoprimo** [comment]
I spoke with @kraftbj from the Jetpack team and this issue is likely related to a Jetpack problem that they already fixed and that will be included in the next release. I was not able to reproduce this problem when running Jetpack release candidate. I'm still waiting to hear when the next version will be released. I'm not sure if we should do something on our side or not (cc @peterfabian).

**Comment by peterfabian** [comment]
This would still give an error if someone is trying to run WC with Jetpack 8.2, right? Then maybe we can add a workaround just for that version? As 8.2 is one of the recent versions, we should not throw errors, in my opinion. Do you know if the manual init is a viable alternative for 8.2 or can it create some problems?

**Comment by kraftbj** [comment]
To confirm, this is happening when Jetpack is not installed too?

**Comment by jeherve** [comment]
> this issue is likely related to a Jetpack problem that they already fixed and that will be included in the next release.

Jetpack 8.2.2 was just released to address those issues.

**Comment by peterfabian** [comment]
For future reference: this issue was actually not solved by JP 8.2.2, we need to apply a fix on our side as well, hence the PR above.
